addons/payment_payulatam/models/payment.py

```diff
@@ -55,7 +55,7 @@ def payulatam_form_generate_values(self, values):
             accountId=self.payulatam_account_id,
             description=values.get('reference'),
             referenceCode=tx.reference,
-            amount=values['amount'],
+            amount=round(values['amount'], 2),
```

Comment:
`from odoo.tools.float_utils import float_compare, float_round`

Check the method's docstring to learn why it's more appropriate than round ;)

Otherwise, LGTM. You can r+ after the change.